### PR TITLE
Only build LLVM targets required for the platform.

### PR DESCRIPTION
**Pull request by pmuetschard**
Significantly reduces libinterceptor.so size on the various Android platforms: 2MB for X86, 4 MB for armeabi and nearly 5MB for aarch64.

### CODE REVIEW

File: tools/build/third_party/llvm/include-linux/llvm/Config/AsmParsers.def [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+|* time, at which point this header is generated. Do not modify this          *|
+|* header directly.                                                           *|
+|*                                                                            *|
+\*===----------------------------------------------------------------------===*/
+
+#ifndef LLVM_ASM_PARSER
+#  error Please define the macro LLVM_ASM_PARSER(TargetName)
+#endif
+
+LLVM_ASM_PARSER(AArch64)
```
Comment (AWoloszyn) [inline tools/build/third_party/llvm/include-linux/llvm/Config/AsmParsers.def]: I am not sure if we need the ASM Parsers for Aarch64/Arm on Linux. (Unless llvm requires them for writing).

File: tools/build/third_party/llvm/include-linux/llvm/Config/Disassemblers.def [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+|* header directly.                                                           *|
+|*                                                                            *|
+\*===----------------------------------------------------------------------===*/
+
+#ifndef LLVM_DISASSEMBLER
+#  error Please define the macro LLVM_DISASSEMBLER(TargetName)
+#endif
+
+LLVM_DISASSEMBLER(AArch64)
+LLVM_DISASSEMBLER(ARM)
```
Comment (AWoloszyn) [inline tools/build/third_party/llvm/include-linux/llvm/Config/Disassemblers.def]: Same as above

File: tools/build/third_party/llvm/include-windows/llvm/Config/AsmParsers.def [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+|* time, at which point this header is generated. Do not modify this          *|
+|* header directly.                                                           *|
+|*                                                                            *|
+\*===----------------------------------------------------------------------===*/
+
+#ifndef LLVM_ASM_PARSER
+#  error Please define the macro LLVM_ASM_PARSER(TargetName)
+#endif
+
+LLVM_ASM_PARSER(AArch64)
```
Comment (AWoloszyn) [inline tools/build/third_party/llvm/include-windows/llvm/Config/AsmParsers.def]: Same for windows, disassembler and parser I don't THINK we need unless LLVM internally requires them for writing out .o files.

File: tools/build/third_party/llvm/include-linux/llvm/Config/AsmParsers.def (hunk 1 above)
Comment (ben-clayton) [reply tools/build/third_party/llvm/include-linux/llvm/Config/AsmParsers.def]: We only need them for lib-interceptor.

File: tools/build/third_party/llvm/include-windows/llvm/Config/AsmParsers.def (hunk 3 above)
Comment (ben-clayton) [reply tools/build/third_party/llvm/include-windows/llvm/Config/AsmParsers.def]: Can we get some stats on binary size reduction and build time reduction? I like happy stats.

Comment (pmuetschard) [reply tools/build/third_party/llvm/include-windows/llvm/Config/AsmParsers.def]: But the stats into the CL and change description.

File: tools/build/third_party/llvm/include-linux/llvm/Config/AsmParsers.def (hunk 1 above)
Comment (pmuetschard) [reply tools/build/third_party/llvm/include-linux/llvm/Config/AsmParsers.def]: Done.

File: tools/build/third_party/llvm/include-linux/llvm/Config/Disassemblers.def (hunk 2 above)
Comment (pmuetschard) [reply tools/build/third_party/llvm/include-linux/llvm/Config/Disassemblers.def]: Done.

**Comment by ben-clayton** [comment]
Looks like it reduced the build time by ~25% too. Nice!

File: tools/build/third_party/llvm/include-linux/llvm/Config/llvm-config.h [hunk 4]
```diff
@@ -32,15 +32,9 @@
...
 #define LLVM_NATIVE_ARCH X86
 
-/* LLVM name for the native AsmParser init function, if available */
-#define LLVM_NATIVE_ASMPARSER LLVMInitializeX86AsmParser
-
 /* LLVM name for the native AsmPrinter init function, if available */
 #define LLVM_NATIVE_ASMPRINTER LLVMInitializeX86AsmPrinter
 
-/* LLVM name for the native Disassembler init function, if available */
-#define LLVM_NATIVE_DISASSEMBLER LLVMInitializeX86Disassembler
```
Comment (ben-clayton) [inline tools/build/third_party/llvm/include-linux/llvm/Config/llvm-config.h]: Hopefully this isn't required for the compiler work...

**Pull request merged**